<a href="https://colab.research.google.com/github/jyouser/Probability-Statistics/blob/main/day_1_Sachin_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Download dataset

In [ ]:
!wget --no-check-certificate https://drive.google.com/uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7 -O Sachin_ODI.csv

--2025-11-02 16:55:02--  https://drive.google.com/uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7
Resolving drive.google.com (drive.google.com)... 173.194.174.139, 173.194.174.102, 173.194.174.101, ...
Connecting to drive.google.com (drive.google.com)|173.194.174.139|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://drive.usercontent.google.com/download?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7 [following]
--2025-11-02 16:55:02--  https://drive.usercontent.google.com/download?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7
Resolving drive.usercontent.google.com (drive.usercontent.google.com)... 142.251.170.132, 2404:6800:4008:c19::84
Connecting to drive.usercontent.google.com (drive.usercontent.google.com)|142.251.170.132|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 26440 (26K) [application/octet-stream]
Saving to: ‘Sachin_ODI.csv’

Sachin_ODI.csv      100%[===================>]  25.82K  --.-KB/s    in 0s      

2025-11-02 16:55:04 (86.0 MB

# Import libraries

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

Read data using pandas

In [ ]:
df=pd.read_csv('Sachin_ODI.csv')
df

runs  NotOut mins   bf  fours  sixes      sr  Inns           Opp  \
0      13       0   30   15      3      0   86.66     1   New Zealand   
1      37       0   75   51      3      1   72.54     2  South Africa   
2      47       0   65   40      7      0  117.50     2     Australia   
3      48       0   37   30      9      1  160.00     2    Bangladesh   
4       4       0   13    9      1      0   44.44     2      Pakistan   
..    ...     ...  ...  ...    ...    ...     ...   ...           ...   
355    14       0   34   15      2      0   93.33     2     Australia   
356    39       0   45   30      5      0  130.00     2     Sri Lanka   
357     6       0   25   19      1      0   31.57     1     Sri Lanka   
358   114       0  205  147     12      1   77.55     1    Bangladesh   
359    52       0   93   48      5      1  108.33     2      Pakistan   

       Ground        Date        Winner    Won  century  
0      Napier  1995-02-16   New Zealand  False    False  
1    Hamilton  1995-02-18  South Africa  False    False  
2     Dunedin  1995-02-22         India   True    False  
3     Sharjah  1995-04-05         India   True    False  
4     Sharjah  1995-04-07      Pakistan  False    False  
..        ...         ...           ...    ...      ...  
355    Sydney  2012-02-26     Australia  False    False  
356    Hobart  2012-02-28         India   True    False  
357     Dhaka  2012-03-13         India   True    False  
358     Dhaka  2012-03-16    Bangladesh  False     True  
359     Dhaka  2012-03-18         India   True    False  

[360 rows x 14 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Data columns (total 14 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   runs     360 non-null    int64  
 1   NotOut   360 non-null    int64  
 2   mins     360 non-null    object 
 3   bf       360 non-null    int64  
 4   fours    360 non-null    int64  
 5   sixes    360 non-null    int64  
 6   sr       360 non-null    float64
 7   Inns     360 non-null    int64  
 8   Opp      360 non-null    object 
 9   Ground   360 non-null    object 
 10  Date     360 non-null    object 
 11  Winner   360 non-null    object 
 12  Won      360 non-null    bool   
 13  century  360 non-null    bool   
dtypes: bool(2), float64(1), int64(6), object(5)
memory usage: 34.6+ KB


Display top 5 rows

In [ ]:
df.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

# Shape of dataset(display rows and columns)

In [ ]:
df.shape

(360, 14)

# Q1. A match is randomly chosen, what is the probability that India have won that match?

In [ ]:
df.loc[df['Won']==True]

runs  NotOut mins   bf  fours  sixes      sr  Inns          Opp  \
2      47       0   65   40      7      0  117.50     2    Australia   
3      48       0   37   30      9      1  160.00     2   Bangladesh   
5     112       1  137  107     15      1  104.67     2    Sri Lanka   
6      41       0   51   41      5      0  100.00     2    Sri Lanka   
8      39       0   79   51      4      0   76.47     2  New Zealand   
..    ...     ...  ...  ...    ...    ...     ...   ...          ...   
349    18       0   21   14      2      0  128.57     2    Sri Lanka   
351    48       0   87   63      5      0   76.19     2    Sri Lanka   
356    39       0   45   30      5      0  130.00     2    Sri Lanka   
357     6       0   25   19      1      0   31.57     1    Sri Lanka   
359    52       0   93   48      5      1  108.33     2     Pakistan   

       Ground        Date Winner   Won  century  
2     Dunedin  1995-02-22  India  True    False  
3     Sharjah  1995-04-05  India  True    False  
5     Sharjah  1995-04-09  India  True     True  
6     Sharjah  1995-04-14  India  True    False  
8    Amritsar  1995-11-18  India  True    False  
..        ...         ...    ...   ...      ...  
349    Mumbai  2011-04-02  India  True    False  
351     Perth  2012-02-08  India  True    False  
356    Hobart  2012-02-28  India  True    False  
357     Dhaka  2012-03-13  India  True    False  
359     Dhaka  2012-03-18  India  True    False  

[184 rows x 14 columns]

In [ ]:
df.loc[df['Won']==True].shape[0]

184

In [ ]:
df.loc[df['Won']==True].shape[0]/df.shape[0]

0.5111111111111111

In [ ]:
(df.loc[df['Won']==True].shape[0]/df.shape[0])*100

51.11111111111111

Conclusion: Probability that India have won that match is 51%

# Q2. A match is chosen at a random, what is the probability that Sachin has scored a Century in that match?

In [ ]:
df.loc[df['century']==True]

runs  NotOut mins   bf  fours  sixes      sr  Inns           Opp  \
5     112       1  137  107     15      1  104.67     2     Sri Lanka   
12    127       1  175  138     15      1   92.02     2         Kenya   
15    137       0  198  137      8      5  100.00     1     Sri Lanka   
20    100       0    -  111      9      1   90.09     1      Pakistan   
23    118       0    -  140      8      2   84.28     1      Pakistan   
27    110       0    -  138      5      1   79.71     1     Sri Lanka   
41    114       0  157  126     14      0   90.47     1  South Africa   
47    104       0  121   97      8      1  107.21     2      Zimbabwe   
55    117       0  188  137     13      2   85.40     2   New Zealand   
85    100       0  132   89      5      7  112.35     2     Australia   
91    143       0  187  131      9      5  109.16     2     Australia   
92    134       0  195  131     12      3  102.29     2     Australia   
95    100       1  156  103     13      0   97.08     2         Kenya   
99    128       0  182  131      8      2   97.70     1     Sri Lanka   
101   127       1  155  130     13      1   97.69     2      Zimbabwe   
104   141       0  191  128     13      3  110.15     1     Australia   
107   118       1  173  112     14      2  105.35     2      Zimbabwe   
110   124       1  140   92     12      6  134.78     2      Zimbabwe   
116   140       1  114  101     16      3  138.61     1         Kenya   
123   120       0  191  141     11      2   85.10     1     Sri Lanka   
128   186       1  221  150     20      3  124.00     1   New Zealand   
143   122       0  195  138     12      0   88.40     2  South Africa   
155   101       0    -  140      3      1   72.14     1     Sri Lanka   
162   146       0  200  153     15      2   95.42     1      Zimbabwe   
167   139       0  197  125     19      0  111.20     1     Australia   
173   122       1  210  131     12      1   93.12     2   West Indies   
175   101       0  209  129      9      0   78.29     1  South Africa   
179   146       0  186  132     17      0  110.60     1         Kenya   
191   105       1  167  108      8      1   97.22     1       England   
194   113       0  141  102     12      1  110.78     1     Sri Lanka   
206   152       0  193  151     18      0  100.66     1       Namibia   
215   100       0  158  119      9      1   84.03     1     Australia   
219   102       0  124   91     12      1  112.08     1   New Zealand   
229   141       0  173  135     17      1  104.44     2      Pakistan   
245   123       0  192  130     12      2   94.61     1      Pakistan   
258   100       0  188  113     10      1   88.49     1      Pakistan   
262   141       1    -  148     13      5   95.27     1   West Indies   
276   100       1  110   76     10      1  131.57     1   West Indies   
313   117       1  235  120     10      0   97.50     2     Australia   
322   163       1  206  133     16      5  122.55     1   New Zealand   
325   138       0  214  133     10      1  103.75     1     Sri Lanka   
331   175       0  210  141     19      4  124.11     2     Australia   
338   200       1  226  147     25      3  136.05     1  South Africa   
342   120       0  170  115     10      5  104.34     1       England   
345   111       0  184  101      8      3  109.90     1  South Africa   
358   114       0  205  147     12      1   77.55     1    Bangladesh   

                 Ground        Date        Winner    Won  century  
5               Sharjah  1995-04-09         India   True     True  
12              Cuttack  1996-02-18         India   True     True  
15                Delhi  1996-03-02     Sri Lanka  False     True  
20            Singapore  1996-04-05      Pakistan  False     True  
23              Sharjah  1996-04-15         India   True     True  
27        Colombo (RPS)  1996-08-28     Sri Lanka  False     True  
41               Mumbai  1996-12-14         India   True     True  
47               Benoni  1997

In [ ]:
df.loc[df['century']==True].shape

(46, 14)

In [ ]:
df.loc[df['century']==True].shape[0]

46

In [ ]:
df.loc[df['century']==True].shape[0]/df.shape[0]

0.12777777777777777

ans is 12.77%

Approach 2:

In [ ]:
df['century'].value_counts()

century
False    314
True      46
Name: count, dtype: int64

In [ ]:
df['century'].value_counts()/df.shape[0]

century
False    0.872222
True     0.127778
Name: count, dtype: float64

ans is 12.77%

Conclusion:If you chose a random match, there is 12.77% chance that Sachin has scored a century in that match

# Q3. A match is chosen at a random. What is the probability that Sachin has scored a century in that match and India have won that match?

Using value_counts() function

In [ ]:
df[['century','Won']].value_counts()

century  Won  
False    False    160
         True     154
True     True      30
         False     16
Name: count, dtype: int64

Using crosstab() function

In [ ]:
# This table is called contingency table
pd.crosstab(index=df['century'],columns=df['Won'],margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

Finding probability of winning and done century

In [ ]:
# Won -> True,and century -> True
#p=count of favourable outcomes/count of ss
(30/360)*100

8.333333333333332

In [ ]:
df[['century','Won']].value_counts()/len(df)

century  Won  
False    False    0.444444
         True     0.427778
True     True     0.083333
         False    0.044444
Name: count, dtype: float64

In [ ]:
sum((df['century']==True) & (df['Won']==True))/len(df)

0.08333333333333333

In [ ]:
(sum((df['century']==True) & (df['Won']==True))/len(df))*100

8.333333333333332

There is 8% chance that Sachin has scored a century and India have won that match

In [ ]:
pd.crosstab(index=df['century'],columns=df['Won'],margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

# Q4. A match is chosen at a random. What is the probability that Sachin has scored a century in that match and India did not won that match?

In [ ]:
(sum((df['century']==True)&(df['Won']==False))/len(df))*100

4.444444444444445

# Conclusion of the Problem statement:

Let's have a look how is Sachin's batting can or cannot impact the winning chances of India

1) Out of the 360 matches that Sachin has played, India have won 184 matches and Loose 176 matches.

2) So, if we choose any match at a random from Sachin's ODI career, there is a 51% chance that India have won that match.

3) Now, If we choose a random match from Sachin's ODI career, there is 12.77% chance that Sachin has scored a century in that match.

4) We know if a random match is choosen, there is 12.77% chance that Sachin has scored a century but there is only 8% chance India have won that match.

we can conclude that the chances of India, Winning a match is more when Sachin didn't score a century (what an amazing insight)

Finally,

We can conclude that, if we pick a random match where Sachin played, India's win percentage is 51%. There is 12.77% chance of Sachin scoring a century in that match, and there is only 8% chance that in that match Sachin scores a century as well as India have won that match.